docs/development/serverless.md

```diff
@@ -0,0 +1,74 @@
+# Serverless
+
+Serverless is a powerful and popular model where you don’t have to worry about managing and maintaining your application infrastructure. There are two main paradigms when it comes to Serverless.  The first is the operations paradigm which is responsible for the orchestration of containers to zero.
+
+The second paradigm is functions.  In the serverless context, a function is a single-purpose piece of code created by the developer but run and monitored by the managed infrastructure. A serverless function’s value is its simplicity and swiftness, which can entice even those who don’t consider themselves developers.
+
+
+## Recommended Components
+
+N/A
+
+## Guidance
+
+While the team doesn't have to much experience in the serverless realm at the moment, there is still some guidance that can be recommended.
+
+### Functions
+
+One of the main recommendations that we can give related to Functions and Serverless in general, is to keep things as stateless as possible.  When a request comes in, the function will scale up and become available for a set period of time before it scales back down and the current context is destroyed.
+
+Node.js is one of the top recommended languages for Functions due to Node's small memory foot print, quick startup time and asynchronus nature.
+
+#### Challenges
+
+There are some challenges when deciding to use functions.  The first is that there is no standard.  The way a function needs to be created and what data is available in the function signature, is based on the vendor that the function is being created for.  While not a standard for how functions should be created, the [CloudEvents specification](https://github.com/cloudevents/spec) is trying to standardize the payload that is sent.
+
+Second is the local development and remote debugging experience.  This can be challenging for a couple of reasons.  First, when developing locally, if the function needs to interact with other remote services, this can be challenging to setup.  Second, once the function is running on its respective serverless platform, debugging the remote function can be challening, since it is no small feat to connect a debugger to a remote service.
```

Comment:
I feel that here, it might be worth mentioning that unit tests are the foundation of functions. Typically, good programming practice is that there should be no side effects from a function call (in any platform/runtime not just functions). If that's the case, then a function should be fully testable for developers using unit tests. This, of course, leaves open integration testing and end to end testing - both of which are still important. But I'm not sure that the challenges in these scenarios are specific to functions. Consider any REST service that runs in a cluster, potentially calling other services - the integration/e2e testing challenges are the same.